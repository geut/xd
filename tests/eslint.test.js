const path = require('path')

const { generateEnvironment } = require('./helpers')
const { invoke } = require('..')

beforeAll(async () => {
  const fixture = path.resolve('./tests/fixtures/eslint')
  this.cwd = await generateEnvironment(fixture, 'xd-eslint-test')
})

test('lint file', () => {
  const args = [`${this.cwd}/index.js`]
  let result = invoke(this.cwd, args, null, 0)
  expect(result).toMatchSnapshot()
  args.push('-f')
  args.push('json')
  result = invoke(this.cwd, args, null, 0)
  expect(result).toMatchSnapshot()
})
