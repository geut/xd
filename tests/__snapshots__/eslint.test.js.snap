// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint: fix file fix --fix -f json 1`] = `"[{\\"filePath\\":\\"/tmp/xd-eslint-test/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"console.log('test');\\\\n\\"}]"`;

exports[`eslint: fix file fix --fix-dry-run 1`] = `"[{\\"filePath\\":\\"/tmp/xd-eslint-test/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"console.log('test');\\\\n\\"}]"`;

exports[`eslint: fix file fix --stdin --fix-dry-run -f json 1`] = `"[{\\"filePath\\":\\"<text>\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"console.log('test');\\\\n\\"}]"`;

exports[`eslint: fix file fix --stdin --stdin-filename "filename" --fix-dry-run -f json 1`] = `"[{\\"filePath\\":\\"/tmp/xd-eslint-test/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"console.log('test');\\\\n\\"}]"`;

exports[`eslint: lint file default lint 1`] = `
"[0m[0m
[0m[4m/tmp/xd-eslint-test/index.js[24m[0m
[0m  [2m1:20[22m  [31merror[39m  Missing semicolon  [2msemi[22m[0m
[0m[0m
[0m[31m[1mâœ– 1 problem (1 error, 0 warnings)[22m[39m[0m
[0m[31m[1m[22m[39m[31m[1m  1 error and 0 warnings potentially fixable with the \`--fix\` option.[22m[39m[0m
[0m[31m[1m[22m[39m[0m
"
`;

exports[`eslint: lint file lint -f json 1`] = `
"[{\\"filePath\\":\\"/tmp/xd-eslint-test/index.js\\",\\"messages\\":[{\\"ruleId\\":\\"semi\\",\\"severity\\":2,\\"message\\":\\"Missing semicolon.\\",\\"line\\":1,\\"column\\":20,\\"nodeType\\":\\"ExpressionStatement\\",\\"fix\\":{\\"range\\":[19,19],\\"text\\":\\";\\"}}],\\"errorCount\\":1,\\"warningCount\\":0,\\"fixableErrorCount\\":1,\\"fixableWarningCount\\":0,\\"source\\":\\"console.log('test')\\\\n\\"}]
"
`;
